app/assets/javascripts/admin/listing_shapes.js

```diff
@@ -93,6 +86,21 @@ window.ST.initializeListingShapeForm = function(formId) {
     toggleLabel($(".js-unit-label"), enabled);
   };
 
+  var addCustomUnitForm = function() {
+    var $form = $(".js-listing-shape-add-custom-unit-form").last().clone();
+    $form.find('input').prop('disabled', false);
+    var randomId = Math.floor(Math.random() * (Number.MAX_VALUE - 1)) + 1;
```

Comment:
If I understand right, this doesn't have to be random, instead, it has to be unique, right?

There's a handy function in lodash `uniq` https://lodash.com/docs#uniq, maybe that could be used here?